pkgs/by-name/fr/freej2me/package.nix

```diff
@@ -0,0 +1,86 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, ant
+, jdk
+, stripJavaArchivesHook
+, SDL2
+, freeimage
+, writeShellScriptBin
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "freej2me";
+  version = "0-unstable-2024-02-16";
+
+  src = fetchFromGitHub {
+    owner = "hex007";
+    repo = "freej2me";
+    rev = "8b9bc8a19baf26e3d92f88934a64a32f1cbc2795";
+    hash = "sha256-FaB2qnbtJnSZbrGRYDiz3Q/uNgD4GX//VaID6jHtyeU=";
+  };
+
+  nativeBuildInputs = [
+    ant
+    jdk
+    stripJavaArchivesHook
+  ];
+
+  buildInputs = [ SDL2 freeimage ];
+
+  prePatch = ''
+    # window may not receive events, which causes issues on fullscreen mode
+    substituteInPlace src/sdl2/anbu.cpp \
+      --replace-fail "SDL_WINDOW_FULLSCREEN_DESKTOP" "SDL_WINDOW_RESIZABLE"
+
+    substituteInPlace src/org/recompile/freej2me/Anbu.java \
+      --replace-fail "/usr/local" "${placeholder "out"}"
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ant
+    make -C src/sdl2
+    make -C src/libretro
+
+    runHook postBuild
+  '';
+
+  preInstall = ''
+    substituteInPlace src/sdl2/Makefile \
+      --replace-fail "/usr/local" "${placeholder "out"}"
+  '';
+
+  installPhase = let
+    script = writeShellScriptBin "freej2me" ''
+      if [ ''$# -ne 4 ]; then
+        echo "usage: freej2me [front-end: (freej2me,freej2me-sdl,freej2me-lr)] [midlet] [width] [height]"
+        exit
+      fi
+
+      ${lib.getExe' jdk "java"} -jar \
+        @out@/share/java/$1.jar "file://$(realpath $2)" $3 $4
+    '';
```

Comment:
Can't we just do wrapper like `java freej2me $@` ? Hard-coding the arguments seems like bad solution to me.

Reply:
See https://github.com/NixOS/nixpkgs/pull/326677/files#r1677273708